### PR TITLE
feat: added like/dislike functionality

### PR DESCRIPTION
**Pull request by coldpigli**

### CODE REVIEW

**Review by vaibhav-mahalle — APPROVED** [review]
Liked the proper formatting of the code, but not much apart from that to review.

File: src/components/PostItem/PostItem.jsx [hunk 1]
```diff
@@ -80,14 +100,16 @@ const PostItem = ({ post }) => {
           {engageButtons.map((item) => (
             <HStack color="#808191" align="center">
               <IconButton
-                variant="ghost"
                 fontSize="1.5rem"
                 icon={item.icon}
+                variant="unstyled"
+                color={item.active ? "#6C5DD3" : "#808191"}
```
Comment (aishwarya-bm) [inline src/components/PostItem/PostItem.jsx:106]: Use color variables

**Review by aishwarya-bm — APPROVED** [review]
Code looks clean and has good naming conventions. Liked the error handling part in redux.

File: src/redux/slices/postSlice.js [hunk 2]
```diff
@@ -32,13 +33,23 @@ const postSlice = createSlice({
...
         },
         [addNewPost.rejected]: (state) => {
             state.postStatus = "failed"
+        },
+        [likeOrDislikePost.pending]: (state) => {
+            state.postStatus = "loading"
+        },
+        [likeOrDislikePost.fulfilled]: (state, action) => {
+            const {posts} = action.payload;
+            console.log("The posts from like Post slice", posts);
```
Comment (sadathshariff) [inline src/redux/slices/postSlice.js:47]: Remove the console logs here and in the handle click as well

**Review by sadathshariff — APPROVED** [review]
Everything looks good, remove the console logs, good to go

**Pull request merged**